### PR TITLE
checking null finger prints for safe identification

### PR DESCRIPTION
**Pull request by amansulaiman**
The current code is throwing an error if some fingers print are null

### CODE REVIEW

File: src/FingerWhoIs.Lib/Native/NativeApi.cs [hunk 1]
```diff
@@ -139,7 +139,14 @@ internal static unsafe NativeCall<List<int>> Indentify(ReadOnlySpan<byte> fmd,Fm
...
-                fmdSizes[i] = fmds[i].Length;
+                if (fmds[i] == null)
+                {
+                    fmdSizes[i] = 0;
+                }
+                else
+                {
+                    fmdSizes[i] = fmds[i].Length;
+                }
             }
```
Comment (Herocod3r) [inline src/FingerWhoIs.Lib/Native/NativeApi.cs]: Can this logic be moved to the FingerEngine Class instead ? the NativeApi serves as a point for marshalling and native calls only, this kind of logic should be done before a call is placed here

**Review by Herocod3r — CHANGES_REQUESTED** [review]

File: src/FingerWhoIs.Lib/Native/NativeApi.cs (hunk 1 above)
Comment (amansulaiman) [reply src/FingerWhoIs.Lib/Native/NativeApi.cs]: Okay

Comment (amansulaiman) [reply src/FingerWhoIs.Lib/Native/NativeApi.cs]: Updated

File: src/FingerWhoIs.Lib/FIngerEngine.cs [hunk 2]
```diff
@@ -61,6 +61,13 @@ public int CompareTwoFmds(ReadOnlySpan<byte> fmd1, ReadOnlySpan<byte> fmd2,FmdFo
         public List<int> IdentifyFmd(ReadOnlySpan<byte> fmd1,ReadOnlySpan<byte[]> fmds,uint maxToReturn = 10,uint threshold = 21474, FmdFormat format = FmdFormat.Iso)
         {
             var fmdsArray = fmds.ToArray();
+            for (int i = 0; i < fmdsArray.Length; i++)
+            {
+                if (fmdsArray[i] == null)
+                {
+                    fmdsArray[i] = new byte[0];
+                }
```
Comment (Herocod3r) [inline src/FingerWhoIs.Lib/FIngerEngine.cs]: Can you use `is` keyword instead ? to check if stuff is null ?, `==` the equality operator might not always give you a desired result.

**Review by Herocod3r — CHANGES_REQUESTED** [review]

**Comment by Herocod3r** [comment]
Thanks @amansulaiman for the PR

**Pull request merged**